### PR TITLE
feat(medusa): Adds shipped status to line items

### PR DESCRIPTION
**Pull request by olivermrbl**
This will add support for automatic captures in individual Medusa projects

### CODE REVIEW

File: packages/medusa/src/services/order.js [hunk 1]
```diff
@@ -655,6 +699,36 @@ class OrderService extends BaseService {
       })
   }
 
+  /**
+   * Registers a fully shipped order.
+   * @param {string} orderId - id of order to capture payment for.
+   * @param {string} details - detailed reason for the failed payment
+   * @return {Promise} result of the update operation.
+   */
+  async registerShipmentStatus(orderId, status) {
```
Comment (srindom) [inline packages/medusa/src/services/order.js:708]: This is here twice

File: packages/medusa/src/services/order.js [hunk 2]
```diff
@@ -418,30 +453,39 @@ class OrderService extends BaseService {
   async createShipment(orderId, fulfillmentId, trackingNumbers, metadata = {}) {
```
Comment (srindom) [inline packages/medusa/src/services/order.js:453]: When does it update the shipment status?

File: packages/medusa/src/services/order.js [hunk 3]
```diff
@@ -253,6 +253,41 @@ class OrderService extends BaseService {
...
+        {
+          _id: order._id,
+        },
+        {
+          fulfillment_status: status,
+        }
+      )
+      .then(result => {
+        // Notify subscribers
+        this.eventBus_.emit(OrderService.Events.PAYMENT_CAPTURE_FAILED, result)
```
Comment (srindom) [inline packages/medusa/src/services/order.js:283]: This shouldn't be `PAYMENT_CAPTURE_FAILED`

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/services/order.js (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/services/order.js:453]: I am not sure what you mean... We don't have a shipment status on orders.

**Review by srindom — APPROVED** [review]
🐑

**Pull request merged**